webrender/res/shared.glsl

```diff
@@ -113,6 +113,22 @@
             return 0.0;
         return 0.5 + dist * (0.8431027 * dist * dist - 1.14453603);
     }
+
+    /// Component-wise selection.
+    ///
+    /// The idea of using this is to ensure both potential branches are executed before
+    /// selecting the result, to avoid observable timing differences based on the condition.
+    ///
+    /// Example usage: color = if_then_else(LessThanEqual(color, vec3(0.5)), vec3(0.0), vec3(1.0));
+    ///
+    /// The above example sets each component to 0.0 or 1.0 independently depending on whether
+    /// their values are below or above 0.5.
+    ///
+    /// This is written as a macro in order to work with vectors of any dimension.
+    ///
+    /// Note: Some older android devices don't support mix with bvec. If we ever run into them
+    /// the only option we have is to polyfill it with a branch per component.
+    #define if_then_else(cond, then_branch, else_branch) mix(else_branch, then_branch, cond)
```

Comment:
d'oh, why don't we just use `step` instead of `lessThanEqual` so that `bvec` isn't involved?
I also don't think we should be having the one-line wrappers around GLSL functions, fwiw

Reply:
I wasn't aware that one can use step with bvec. It's not clear from the [glsl docs](https://www.khronos.org/registry/OpenGL-Refpages/gl4/html/step.xhtml) that you could do that whereas it is explicitly specified for [mix](https://www.khronos.org/registry/OpenGL-Refpages/gl4/html/mix.xhtml).
We do want component-wise selection here so bvec is needed.

I stand by wrapping the single line function because the function will most likely be used all over the place to avoid timing attacks, using mix directly is error prone (the bug this PR is fixing) and this might need to be polyfilled if we support low end android devices.

Reply:
What I'm saying is: we don't need `bvec` at all. We can just use `mix` and `step` without triggering the bugtrap GLSL booleans are.

Reply:
Not sure I follow, you want to do it manually for each individual channel?

Reply:
I'm saying that instead of doing this:
```glsl
mix(c1, c2, lessThanEqual(color, vec3(0.04045)));
```
we could be doing this:
```glsl
mix(c1, c2, step(color, vec3(0.04045)));
```
And it would not need any wrappers or messing around `bvec`